### DISCUSSION TITLE
Problema con Task 3: sklearn compatible class

### DISCUSSION THREAD

**Discussion by Bexzho** [rramosp/ai4eng.v1#13] [05 - GESTIÓN DE MODELOS]
Hola, no entendí de forma clara la creación del compatible class with your model. He intentado organizar mi código con las funciones antes creadas en el mismo LAB siguiendo las indicaciones, pero siempre obtengo problemas.
Adjunto imagen de lo que he hecho:

![1](https://user-images.githubusercontent.com/58709463/119270236-4ffd1380-bbc1-11eb-8d93-ba2d85780ca7.png)
 
Lo que no entiendo es el *self* que se introduce en las funciones y como usarla dentro de ellas.

¿Me podrían comentar que es lo que debo arreglar en el código para que funcione? 

Gracias.

**Comment by rramosp** [comment]
Hola Alexander, tienes que mirar un poco cómo funciona el mecanismo de orientación a objetos en Python.

Te recomiendo este enlace si quieres ver los detalles: https://realpython.com/python3-object-oriented-programming/

Pero en breve, fíjate que normalmente (1) se crea una instancia de la clase SimpleModel; (2) se llama al fit con X,y; (3) se llama al predict con X (o con un X nuevo).

El resultado del fit es el mismo del ejercicio anterior: unos valores concretos de theta. Predict tiene que usar los valores que ha encontrado tu fit.

Para esto: 

- al final de tu `fit` tienes que guardar los `thetas` en `self` ... por ejemplo `self.t = thetas` 
- en `predict` tienes que usar los thetas guardados anteriormente. Es decir, tienes que usar `self.t` en lugar de `t`

por otro lado, tienes que tener cuidado que el `return` en el `predict` lo tienes dentro del  `for` y probablemente lo quieras tener fuera (cuando acabe el `for`)

saludos
raúl

**Reply by Bexzho** [reply]
Profe, muchas gracias.
Ya hice lo que usted me sugirió y funcionó. Además voy a revisar la información que me compartió.
